### PR TITLE
feat(trading): order details dialog

### LINKED ISSUE #3394: Single order view
See also: https://github.com/vegaprotocol/frontend-monorepo/issues/4098

## Story

As a user
I want a way to view all information relating to an order
So that I can inspect the order and trades it created

## Draft Acceptance Criteria

To be refined based on UX Exploration

- [x] I can click an order in the order datagrid and be taken to some UI which shows further details about the order
- [x] I can see all the order data
- [ ] ~~I can see all fills/trades created from the order~~ ([Barney and Matt agreed to omit this]
- [x] I can see if the order was amended and when
- [ ] ~~I can link to the corresponding transaction in explorer~~ (can't do this without the transaction id that we don't have at 
- [ ] ~~I can link to the corresponding transaction in explorer for each order amendment~~ 
- [ ] ~~I can link to the corresponding transaction in explorer for an order cancellation~~ 
(checked with Edd and unfortunately we can't do that without having the transaction IDs attached to the order object)
- [x] I can see if and when the order was cancelled 
- [x] There is a way to get back to the order datagrid
- [x] I can see iceberg order data

## Tasks

- [x] UX Exploration (Incl. sketching and creating / refining tickets to represent the work)
- [x] Team and stakeholder review and refine
- [ ] Add/Update specs
- [x] Visual design
- [x] Implementation
- [x] Write tests (unit and e2e) 
- [x] Code review
- [ ] Verify acceptance criteria met
- [x] Exploratory testing to find bugs, edge cases and verify adherence to visual design
- [ ] Write automated regression tests for the A/Cs and add to spec doc (expanding on A/Cs where possible)

### PR DESCRIPTION
**Pull request by MadalinaRaicu**
# Related issues 🔗

Closes #3394 #4098 

# Description ℹ️

View order details and iceberg details.

# Demo 📺
Screenshot 2023-08-01 at 17 59 46

### CODE REVIEW

File: libs/orders/src/lib/components/order-list/order-list.tsx [hunk 1]
```diff
@@ -283,7 +298,20 @@ export const OrderListTable = memo<
                       </ButtonLink>
                     </>
                   )}
-                  <OrderActionsDropdown id={data?.id} />
+                  <ActionsDropdown data-testid="market-actions-content">
+                    <DropdownMenuCopyItem
```
Comment (mattrussell36) [inline libs/orders/src/lib/components/order-list/order-list.tsx:302]:
```suggestion
                    <TradingDropdownCopyItem
```

File: libs/orders/src/lib/components/order-list/order-list.tsx [hunk 2]
```diff
@@ -283,7 +298,20 @@ export const OrderListTable = memo<
...
+                      value={data.id}
+                      text={t('Copy order ID')}
+                    />
+                    <DropdownMenuItem
```
Comment (mattrussell36) [inline libs/orders/src/lib/components/order-list/order-list.tsx:306]:
```suggestion
                    <TradingDropdownItem
```

File: libs/orders/src/lib/components/order-list/order-view-dialog.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+import {
+  MarketState,
+  MarketTradingMode,
+  OrderStatus,
+  OrderTimeInForce,
+  OrderType,
+  Side,
+} from '@vegaprotocol/types';
+
+jest.mock('@vegaprotocol/react-helpers', () => ({
```
Comment (mattrussell36) [inline libs/orders/src/lib/components/order-list/order-view-dialog.spec.tsx:14]: Why do we need to mock here?

File: libs/orders/src/lib/components/order-list/order-view-dialog.tsx [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+  onChange,
+}: OrderViewDialogProps) => {
+  const [, setCopied] = useCopyTimeout();
+  return (
+    <Dialog open={isOpen} title={t('Order details')} onChange={onChange}>
+      <KeyValueTable>
+        <KeyValueTableRow key={'order-market'}>
+          <div data-testid={'order-market-label'}>{t('Market')}</div>
+          <div data-testid={`order-market-value`}>
+            <Link to={`/markets/${order.market?.id}`}>
```
Comment (mattrussell36) [inline libs/orders/src/lib/components/order-list/order-view-dialog.tsx:37]: Use the `Links[Routes.MARKET](order.market.id)` from client-router.tsx

File: libs/orders/src/lib/components/order-list/order-view-dialog.tsx [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+          </KeyValueTableRow>
+        )}
+        <KeyValueTableRow key={'order-id'}>
+          <div data-testid={'order-id-label'}>{t('Order ID')}</div>
+          <div data-testid={`order-id-value`}>
+            {truncateMiddle(order.id, 10)}
+            <CopyToClipboard text={order.id} onCopy={() => setCopied(true)}>
+              <button
+                type="button"
+                data-testid="copy-vega-public-key"
```
Comment (mattrussell36) [inline libs/orders/src/lib/components/order-list/order-view-dialog.tsx:108]: This test-id is inaccurate

File: libs/orders/src/lib/components/order-list/order-view-dialog.tsx [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+                <span className="sr-only">{t('Copy')}</span>
+                <VegaIcon name={VegaIconNames.COPY} />
+              </button>
+            </CopyToClipboard>
+          </div>
+        </KeyValueTableRow>
+        <KeyValueTableRow key={'order-created'}>
+          <div data-testid={'order-created-label'}>{t('Created')}</div>
+          <div data-testid={`order-created-value`}>
+            {new Date(order.createdAt).toLocaleString()}
```
Comment (mattrussell36) [inline libs/orders/src/lib/components/order-list/order-view-dialog.tsx:120]: Why not use our Intl object?

`getDateTimeFormat().format(...)`

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: libs/orders/src/lib/components/order-list/order-view-dialog.spec.tsx (hunk 3 above)
Comment (MadalinaRaicu) [reply libs/orders/src/lib/components/order-list/order-view-dialog.spec.tsx:14]: for the `useCopyTimeout` hook

File: libs/orders/src/lib/components/order-list/order-view-dialog.tsx [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+          <div data-testid={`order-type-value`}>
+            {Schema.OrderTypeMapping[order.type as Schema.OrderType]}
+          </div>
+        </KeyValueTableRow>
+        <KeyValueTableRow key={'order-price'}>
+          <div data-testid={'order-price-label'}>{t('Price')}</div>
+          <div data-testid={`order-price-value`}>
+            {addDecimalsFormatNumber(
+              order.price,
+              order.market?.decimalPlaces as number
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/orders/src/lib/components/order-list/order-view-dialog.tsx:53]: hmm - decimalPlaces can be 0

File: libs/orders/src/lib/components/order-list/order-list.tsx (hunk 2 above)
Comment (MadalinaRaicu) [reply libs/orders/src/lib/components/order-list/order-list.tsx:306]: updated

File: libs/orders/src/lib/components/order-list/order-view-dialog.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+
+    await act(async () => {
+      render(
+        <BrowserRouter>
+          <OrderViewDialog order={order} onChange={jest.fn()} isOpen={true} />
+        </BrowserRouter>
+      );
+    });
+
+    expect(await screen.getByTestId('order-market-label')).toHaveTextContent(
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/orders/src/lib/components/order-list/order-view-dialog.spec.tsx]: updated

File: libs/orders/src/lib/components/order-list/order-view-dialog.spec.tsx [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+      icebergOrder: {
+        __typename: 'IcebergOrder',
+        peakSize: '5000',
+        minimumVisibleSize: '2000',
+        reservedRemaining: '5000',
+      },
+      __typename: 'Order',
+    };
+
+    await act(async () => {
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/orders/src/lib/components/order-list/order-view-dialog.spec.tsx]: updated

File: libs/orders/src/lib/components/order-list/order-view-dialog.spec.tsx (hunk 3 above)
Comment (MadalinaRaicu) [reply libs/orders/src/lib/components/order-list/order-view-dialog.spec.tsx:14]: otherwise the test doesn't run?

File: libs/orders/src/lib/components/order-list/order-view-dialog.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/orders/src/lib/components/order-list/order-view-dialog.tsx]: I will check if it has `order.market`

File: libs/orders/src/lib/components/order-list/order-view-dialog.tsx (hunk 6 above)
Comment (MadalinaRaicu) [reply libs/orders/src/lib/components/order-list/order-view-dialog.tsx]: updated

File: libs/orders/src/lib/components/order-list/order-view-dialog.tsx (hunk 5 above)
Comment (MadalinaRaicu) [reply libs/orders/src/lib/components/order-list/order-view-dialog.tsx]: updated

File: libs/orders/src/lib/components/order-list/order-view-dialog.tsx (hunk 4 above)
Comment (MadalinaRaicu) [reply libs/orders/src/lib/components/order-list/order-view-dialog.tsx]: updated

File: libs/orders/src/lib/components/order-list/order-view-dialog.spec.tsx (hunk 3 above)
Comment (mattrussell36) [reply libs/orders/src/lib/components/order-list/order-view-dialog.spec.tsx:14]: Runs fine for me. Could you double check

**Review by mattrussell36 — APPROVED** [review]
